pkg/k8s/client.go

```diff
@@ -0,0 +1,92 @@
+package k8s
+
+import (
+	"context"
+
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	k8sclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Perform a compile time check.
+var _ Client = &KubeClient{}
+
+const DestinationRuleCrdName string = "destinationrules.networking.istio.io"
+
+//go:generate mockery --name=Client --outpkg=mocks --case=underscore
+type Client interface {
+	PatchApply(context.Context, *unstructured.Unstructured) error
+	GetStatefulSet(context.Context, string, string) (*appsv1.StatefulSet, error)
+	Delete(context.Context, *unstructured.Unstructured) error
+	GetSecret(context.Context, string, string) (*apiv1.Secret, error)
+	GetCRD(context.Context, string) (*apiextensionsv1.CustomResourceDefinition, error)
+	DestinationRuleCRDExists(context.Context) (bool, error)
+}
+
+type KubeClient struct {
```

Comment:
Tests for this client will be added in Next PR. Added a todo [here](https://github.com/kyma-project/nats-manager/issues/11#issue-1631821460).